platform/wayland/cog-view-wl.c

```diff
@@ -301,8 +297,12 @@ void
 cog_wl_view_resize(CogWlView *view)
 {
     CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
-    CogWlViewport *viewport = COG_WL_VIEWPORT(platform->viewport);
+    g_assert(platform);
 
+    if (!platform->display || !platform->display->current_output)
+        return;
+
+    CogWlViewport *viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
```

Comment:
Another one: `g_autoptr` for the `viewport` pointer and a null-check.

I think we may want to do a follow-up patch later in a separate PR to do all the null-checking on the viewport, but at least let's make sure we use `g_autoptr` for the pointers returned by `cog_view_get_viewport()` to avoid leaks.

Reply:
Hopefully I added now all the null checks required.